plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php

```diff
@@ -115,18 +115,32 @@
 
 		}
 
-		woocommerce_wp_select(
-			array(
-				'id'            => '_stock_status',
-				'value'         => $product_object->get_stock_status( 'edit' ),
-				'wrapper_class' => 'stock_status_field hide_if_variable hide_if_external hide_if_grouped',
-				'label'         => __( 'Stock status', 'woocommerce' ),
-				'options'       => wc_get_product_stock_status_options(),
-				'desc_tip'      => true,
-				'description'   => __( 'Controls whether or not the product is listed as "in stock" or "out of stock" on the frontend.', 'woocommerce' ),
-			)
+		$stock_status_options     = wc_get_product_stock_status_options();
+		$common_stock_status_args = array(
+			'id'            => '_stock_status',
+			'value'         => $product_object->get_stock_status( 'edit' ),
+			'wrapper_class' => 'stock_status_field hide_if_variable hide_if_external hide_if_grouped',
+			'label'         => __( 'Stock status', 'woocommerce' ),
+			'options'       => $stock_status_options,
 		);
 
+		/**
+		 * Allow 3rd parties to control whether the "Stock status" option will use radio buttons or a select.
+		 *
+		 * @since 7.6.0
+		 *
+		 * @param bool If false, the "Stock status" will be shown as a select. Default: it will use radio buttons.
+		 */
+		if ( apply_filters( 'woocommerce_stock_status_use_radio', count( $stock_status_options ) === 3 ) ) {
```

Comment:
We probably want to use radio if the count is 2 or 3?

Reply:
Nice catch! I think that less than 4 and more than 0 would work here. Fixed in commit [85503ef](https://github.com/woocommerce/woocommerce/pull/37278/commits/85503ef455e9ac2a3d25e52e707edd4f19892116)

Reply:
True, in the case where there is only one option, either the dropdown or single radio button is equally odd looking, so I'm cool with this!